can/interfaces/socketcand/socketcand.py

```diff
@@ -27,11 +27,12 @@ def convert_ascii_message_to_can_message(ascii_msg: str) -> can.Message:
         frame_string = ascii_msg[8:-2]
         parts = frame_string.split(" ", 3)
         can_id, timestamp = int(parts[0], 16), float(parts[1])
+        is_ext = (len(parts[0]) > 3) or (can_id > 0x7FF)
```

Comment:
The check for `can_id > 0x7FF` is a very common wrong assumption.

Shouldn't it be: `is_ext = (len(parts[0]) == 8)` ??

Reply:
I tried to observe the robustness principle. This part of the code is parsing the ASCII message from socketcand. Sure, from what I have observed, extended messages should be 8 chars. However, what if due to a bug or a slight change in behavior, a message with an arbitration id > 0x7FF but less than 8 chars is received? I would say instead of dropping the frame altogether, interpret it as an extended frame.

Reply:
As there can be EXT frames with CAN IDs 0x000 - 0x7FF too, looking at the CAN ID is simply wrong.

If you want to be more robust the check could be ` is_ext = (len(parts[0]) != 3)` .

Additionally there is no way for a _slight change_ in behavior about the length being either 3 or 8.
Believe me! I wrote that code :-D

Reply:
Sounds good - that's exactly the type of feedback I was looking for. Force pushed and adopted the != 3 method.

Also, please take a look at #1505 while you're at it.

Reply:
What do you think about using len(parts[0]) >3 instead of != 3?

Reply:
> What do you think about using len(parts[0]) >3 instead of != 3?

In fact I would suggest != 3 as there are only 3 or 8. And when it is not 3 , it is definitely safe to store the other stuff as EXT CAN ID - even when it would be an error (that should be impossible).

Reply:
Ok, in that case this is ready to go. Thanks